server/issue.go

```diff
@@ -168,18 +168,44 @@ func httpAPICreateIssue(ji Instance, w http.ResponseWriter, r *http.Request) (in
 		return http.StatusInternalServerError, errors.Errorf("Failed to create issue. %s", err.Error())
 	}
 
-	// Reply to the post with the issue link that was created
+	// Reply with an ephemeral post with the Jira issue formatted as slack attachment.
 	reply := &model.Post{
 		Message:   fmt.Sprintf("Created a Jira issue %v/browse/%v", ji.GetURL(), created.Key),
 		ChannelId: channelId,
 		RootId:    rootId,
 		ParentId:  rootId,
+		UserId:    ji.GetPlugin().getConfig().botUserID,
+	}
+
+	attachment, err := ji.GetPlugin().getIssueAsSlackAttachment(ji, jiraUser, created.Key)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to create notification post "+create.PostId)
+	}
+
+	reply.AddProp("attachments", attachment)
+	_ = api.SendEphemeralPost(mattermostUserId, reply)
+
+	// Fetching issue details as Jira only returns the issue id and issue key at the time of
+	// issue creation. We will not have issue summery in the creation response.
+	createdIssue, err := client.GetIssue(created.Key, nil)
+	if err != nil {
+		return http.StatusInternalServerError,
+			errors.WithMessage(err, "failed to fetch issue details "+created.Key)
+	}
+
+	// Create a public post for all the channel members
+	publicReply := &model.Post{
+		Message:   fmt.Sprintf("Created a Jira issue: %s", mdKeySummaryLink(createdIssue)),
+		ChannelId: channelId,
+		RootId:    rootId,
+		ParentId:  rootId,
 		UserId:    mattermostUserId,
 	}
-	_, appErr = api.CreatePost(reply)
-	if appErr != nil {
+	_, postErr := api.CreatePost(publicReply)
```

Comment:
3/5 This seems nit-picky, but can you keep the err named `appErr`. This is the naming convention we use for errors with the `*model.AppError` type

Reply:
Updated.